Add mention of geo_shape support in geotile and geohash grid agg docs (#61129)
elastic/elasticsearch: docs/reference/aggregations/bucket/geohashgrid-aggregation.asciidoc

<titleabbrev>Geohash grid</titleabbrev>
++++

> A multi-bucket aggregation that works on `geo_point` and `geo_shape` fields and groups points, lines and shapes into buckets that represent cells in a grid.

Reviewer: I tried to streamline this a little bit, but feel free to ignore if wanted.
```suggestion
A multi-bucket aggregation that groups <<geo-point,`geo_point`>> and
<<geo-shape,`geo_shape`>> values into buckets that represent a grid.
```

WARNING: The highest-precision geohash of length 12 produces cells that cover less than a square metre of land and so high-precision requests can be very costly in terms of RAM and result sizes.
Please see the example below on how to first filter the aggregation to a smaller geographic area before requesting high-levels of detail.

> The specified field must be of type `geo_point` or `geo_shape` (which can only be
> set explicitly in the mappings). It can also hold an array of `geo_point` fields,
> in which case all points will be taken into account during aggregation.

Reviewer: I wonder if this paragraph is needed. We already state that the agg works on `geo_point` and `geo_shape` fields. The note about explicit mappings seems more relevant to the field types than the agg.

Do users commonly use `geo_point` to store arrays? If so, would they assume array values are supported (as they are elsewhere in ES)?

I included a suggestion if we decide to keep the para. Feel free to ignore if wanted.
```suggestion
You can only use `geotile_grid` to aggregate an explicitly mapped `geo_point` or
`geo_shape` field. If the `geo_point` field contains an array, `geotile_grid`
aggregates all the array values.
```

WARNING: The highest-precision geohash of length 12 produces cells that cover less than a square metre of land and so high-precision requests can be very costly in terms of RAM and result sizes.
Please see the example below on how to first filter the aggregation to a smaller geographic area before requesting high-levels of detail.

> You can only use `geotile_grid` to aggregate an explicitly mapped `geo_point` or
> `geo_shape` field. If the `geo_point` field contains an array, `geotile_grid`
> aggregates all the array values.

Reviewer: Fixes a copy/paste error on the agg name
```suggestion
You can only use `geohash_grid` to aggregate an explicitly mapped `geo_point` or
`geo_shape` field. If the `geo_point` field contains an array, `geohash_grid`
aggregates all the array values.
```

Author: 😄

Author: was too haste over here. good catch
